icicle/include/icicle/vec_ops.h

```diff
@@ -46,6 +46,37 @@ namespace icicle {
     return config;
   }
 
+  // Reduction operations
+
+  /**
+   * @brief Computes the product of all elements in a vector.
+   *
+   * @tparam T Type of the elements in the vector.
+   * @param vec_a Input vector.
+   * @param n Number of elements in the vector.
+   * @param config Configuration for the operation.
+   * @param output Output scalar to store the result.
+   * @return eIcicleError Error code indicating success or failure.
+   */
+
+  template <typename T>
+  eIcicleError vector_product(const T* vec_a, uint64_t n, const VecOpsConfig& config, T* output, uint64_t offset, uint64_t  stride);
```

Comment:
The order of params should be consistent with other APIs: first inputs and non-optional params, then config, then output. You need to move offset and stride to be before the config (see slice as reference).
In addition, maybe offset is unnecessary since a user can easily do it with pointer arithmetics.

Reply:
Perfect suggestion! Let's first figure out how to use `vec_ops.h` in c++ examples. Next I will reorder parameters accordingly.